src/sentry/rules/processing/delayed_processing.py

```diff
@@ -200,38 +203,85 @@ def parse_rulegroup_to_event_data(
     return parsed_rulegroup_to_event_data
 
 
+def bulk_fetch_events(event_ids: list[str], project_id: int) -> dict[str, Event]:
+    event_limit = 100
+    node_id_to_event_id: dict[str, str] = {
+        Event.generate_node_id(project_id, event_id=event_id): event_id for event_id in event_ids
+    }
+    node_ids = list(node_id_to_event_id.keys())
+    fetch_retry_policy = ConditionalRetryPolicy(should_retry_fetch, exponential_delay(1.00))
+
+    for node_id_chunk in chunked(node_ids, event_limit):
+        bulk_data = fetch_retry_policy(lambda: nodestore.backend.get_multi(node_id_chunk))
```

Comment:
You need to define the dict, and do `bulk_data.update(...)` here, otherwise you're just overwriting results